tests/test_nn/test_modules.py

```diff
@@ -390,6 +391,36 @@ def _additional_score_checks(self, scores):
         pass
 
 
+class MonotonicAffineTransformationInteractionTests(cases.InteractionTestCase):
+    """Tests for monotonic affine transformation interaction adapter."""
+
+    cls = pykeen.nn.modules.MonotonicAffineTransformationInteraction
+    kwargs = dict(
+        base=pykeen.nn.modules.TransEInteraction(p=2),
+    )
+
+    def test_forward_consistency_with_functional(self):  # noqa: D102
```

Comment:
would it make sense to make the interaction test case into a hierarchy of classes? or is this overkill?

Reply:
We could add an intermediate `FunctionalInteractionTests` class.

Then again, while currently not in the code-base (afaik), we could also have a translational interaction function, which is not functional (the simplest example being a chain of TransE (stateless) + AffineTransformation (global state)). Thus, we cannot easily subclass `TranslationalInteractionTests` from `FunctionalInteractionTests`.

Reply:
Maybe we can solve this via [composition](https://python-patterns.guide/gang-of-four/composition-over-inheritance/) instead of inheritance?

Reply:
oOoOoO fancy

Reply:
Unnecessary commentary: Raymond H gave a great talk that covered using composition as a way to write better mocks for testing (https://www.youtube.com/watch?v=EiOglTERPEo)